src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java

```diff
@@ -165,6 +165,45 @@ public <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToCQL(
     return match;
   }
 
+  /**
+   * Returns a codec for the key of the map column.
+   *
+   * @param table {@link TableMetadata} to find the column definition in.
+   * @param column {@link CqlIdentifier} target map column.
+   * @param key The json map key to be used to find the cql codec.
+   * @param <JavaT> Type of the Java object we want to convert.
+   * @param <CqlT> Type fo the Java object the CQL driver expects.
+   * @throws UnknownColumnException
+   * @throws MissingJSONCodecException
+   * @throws ToCQLCodecException
+   */
+  public <JavaT, CqlT> JSONCodec<JavaT, CqlT> getKeyCodecForMap(
+      TableMetadata table, CqlIdentifier column, Object key)
+      throws UnknownColumnException, MissingJSONCodecException, ToCQLCodecException {
+    Objects.requireNonNull(table, "table must not be null");
+    Objects.requireNonNull(column, "column must not be null");
+    var columnMetadata =
+        table.getColumn(column).orElseThrow(() -> new UnknownColumnException(table, column));
+
+    var mapType = columnMetadata.getType();
+    if (!(mapType instanceof MapType)) {
+      throw new ToCQLCodecException(key, mapType, "column is not a map column");
+    }
+    if (key == null) {
+      throw new ToCQLCodecException(
+          null, mapType, "null keys/values are not allowed in map column");
+    }
+    var keyType = ((MapType) mapType).getKeyType();
```

Comment:
nit: there is an if statement above that is checking the type, and then cast again here. Can we make the if on line 189 and this line on 196 work together better ?